server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -1835,6 +1838,9 @@ public Boolean doInTransaction(final TransactionStatus status) {
                     _vlanDao.remove(vlan.getId());
                 }
 
+                // delete template refs for this zone
+                templateZoneDao.deleteByZoneId(zoneId);
```

Comment:
@DaanHoogland please review if this is the correct place to cleanup template_zone_ref entries for a zone to be removed? Or should this call be made after the successful removal of DataCenterVO in success block below?
cc @rhtyd

Reply:
@shwstppr absolutely only on success. I don't think the zone is still usable if deletion of the zoneVO, but it is not right to do an automatic clean-up of items (templates) in a container (zone) if the container is not disposed of.